src/test/java/bio/terra/workspace/integration/WorkspaceIntegrationTest.java

```diff
@@ -28,24 +29,91 @@
 @TestPropertySource("classpath:application-integration-test.properties")
 public class WorkspaceIntegrationTest {
 
-  @Autowired private WorkspaceManagerTestClient workspaceManagerTestClient;
+  // TODO: As this class grows, consider if it's worth breaking down these workspace tests into
+  //  different class files based on the the type of workspace action (Create, Get, Delete, etc).
 
+  @Autowired private WorkspaceManagerTestClient workspaceManagerTestClient;
   @Autowired private TestUtils testUtils;
-
   @Autowired private TestConfiguration testConfig;
-
   @Autowired private AuthService authService;
+  private final String TAG_NEEDS_CLEANUP = "needs-cleanup";
+  private static final Logger logger = LoggerFactory.getLogger(WorkspaceIntegrationTest.class);
+
+  // Variable for workspace Id's to clean up in tearDown(TestInfo testInfo) method, after
+  // running applicable test method(s)
+  private UUID workspaceIdToCleanup;
 
   @BeforeEach
   public void setup() {}
 
+  /*
+   * TODO: Cloud Recourse Library (CRL) is planning to broadly address resource cleanup for integration tests in MC Terra
+   *  applications.
+   *  The doc below proposes a "Resource Tracking" approach for cleaning up. We should follow the developments in CRL for
+   *  potential utilization here in Workspace Manager.
+   *  Doc: https://docs.google.com/document/d/13mYVJML_fOLsX1gxQxRJgECUNT27dAMKzEJxUEvtQqM/edit#heading=h.x06ofvfgp7wt
+   */
+  @AfterEach
+  public void tearDown(TestInfo testInfo) throws Exception {
+    Set<String> tags = testInfo.getTags();
+    if (tags != null && tags.contains(TAG_NEEDS_CLEANUP)) {
+      cleanUpWorkspace();
```

Comment:
assuming we move to tracking all workspaces-to-be-cleaned in a `ConcurrentHashMap`, this should calculate test-specific info and pass it into `cleanUpWorkspace()` - either pass the test name or pass the list of ids to clean up.